File: src/lecture06/java/Main.java
package lecture06.java;

import java.util.Arrays;
import java.util.List;

public class Main {

    public static void main(String[] args) {
        // for-each
        List<Long> numbers = Arrays.asList(1L, 2L, 3L);
        for (Long number : numbers) {
            System.out.println(number);
        }

        // for(up)
        for (int i = 1; i <= 3; i++) {
            System.out.println(i);
        }

        // for(down)
        for (int i = 3; i >= 1; i--) {
            System.out.println(i);
        }

        // for(step)
        for (int i = 1; i <= 5; i += 2) {
            System.out.println(i);
        }

        // while
        int i = 1;
        while (i <= 3) {
            System.out.println(i++);
        }
    }
}

File: src/lecture06/kotlin/Main.kt
package lecture06.kotlin

fun main() {
    // for-each
    val numbers = listOf(1L, 2L, 3L)
    for (number in numbers) {
        println(number)
    }

    // for(up)
    for (i in 1..3) {
        println(i)
    }

    // for(down)
    for (i in 3 downTo 1) {
        println(i)
    }

    // for(step)
    for (i in 1..5 step 2) {
        println(i)
    }

    // while
    var i = 1;
    while (i <= 3) {
        println(i++)
    }
}
